app/src/main/res/layout/fragment_new_profile.xml

```diff
@@ -24,115 +24,121 @@
+                                android:layout_height="wrap_content"
+                                android:layout_weight=".1" />
+
+                        <EditText
+                                android:id="@+id/EndTime"
+                                android:layout_width="0dp"
+                                android:layout_height="wrap_content"
+                                android:layout_weight=".35"
+                                android:focusable="false"
+                                android:focusableInTouchMode="false"
+                                android:hint="@string/end_time"
+                                android:padding="10dp"
+                                android:textAlignment="center"
+                                android:textIsSelectable="false" />
+
+                        <TextView
+                                android:layout_width="0dp"
+                                android:layout_height="wrap_content"
+                                android:layout_weight=".1" />
+
+                    </LinearLayout>
+
+                    <LinearLayout
+                            android:layout_width="match_parent"
                             android:layout_height="wrap_content"
-                            android:layout_weight=".1" />
-
-                </LinearLayout>
+                            android:layout_marginStart="35dp"
+                            android:layout_marginTop="50dp"
+                            android:layout_marginEnd="35dp"
+                            android:visibility="visible">
 
-                <LinearLayout
-                        android:layout_width="match_parent"
-                        android:layout_height="wrap_content"
-                        android:layout_marginStart="35dp"
-                        android:layout_marginTop="50dp"
-                        android:layout_marginEnd="35dp"
-                        android:visibility="visible">
+                        <ca.antonious.materialdaypicker.MaterialDayPicker
+                                android:id="@+id/day_picker"
+                                android:layout_width="match_parent"
+                                android:layout_height="wrap_content" />
 
-                    <ca.antonious.materialdaypicker.MaterialDayPicker
-                            android:id="@+id/day_picker"
-                            android:layout_width="match_parent"
-                            android:layout_height="wrap_content" />
-                </LinearLayout>
+                    </LinearLayout>
 
-            </LinearLayout>
+                    <androidx.appcompat.widget.AppCompatButton
+                            android:id="@+id/makeProfileFab"
+                            android:layout_width="wrap_content"
+                            android:layout_height="wrap_content"
+                            android:layout_gravity="bottom|center"
+                            android:layout_marginTop="180dp"
```

Comment:
Why hardcode the marginTop value?
The more you hardcode the values the more inconsistent it looks on different devices.

Reply:
See, earlier there was no scroll view so we were able to align the button at bottom and add margin_bottom.
But as we are now using scroll view, our button comes inside it. We cannot align it at bottom, because it goes at bottom of scroll view.

And even if we align it at bottom of scroll view, we cannot get the button as expected. Because height of scroll view is "wrap_content"

Reply:
if we set scroll view as match_parent it will overlap action bar

Reply:
To avoid such errors i was using different layout files.

Reply:
Are you sure that this issue can't be fixed using a single layout file? If the action bar becomes part of scrollview then can you fix this issue?

Reply:
but I don't think in apps, action bar goes up on scrolling contents. It is fixed

Reply:
check the youtube app.

Reply:
Oh yeah, my bad. Shall I implement it?

Reply:
Yes

Reply:
@rob729 it is not coming as expected. Because even on setting liner layout(inside scroll view) height as match_parent it is not covering complete length of the screen.
Due to this, either I have to hardcode margin_top for button or use another layout file.

Reply:
Let me know how to proceed

Reply:
Search online for this problem. It has to do with the scrollview. You can make scrollview cover the whole screen.

Reply:
Yes i can but for that i have to set top margin

Reply:
TopMargin for the button?

Reply:
Yes

Reply:
Use relative layout  inside the scrollview